src/sirocco/parsing/_yaml_data_models.py

```diff
@@ -376,6 +380,8 @@ class ConfigData(BaseModel):
 
 
 def get_plugin_from_named_base_model(data: dict) -> str:
+    if isinstance(data, (ConfigRootTask, ConfigShellTask, ConfigIconTask)):
+        return data.plugin
```

Comment:
This is for the idempotence right? Maybe we then also change the input type data to 
```suggestion
def get_plugin_from_named_base_model(data: dict | type[ConfigRootTask | ConfigShellTask | ConfigIconTask]) -> str:
    if isinstance(data, (ConfigRootTask, ConfigShellTask, ConfigIconTask)):
        return data.plugin
```

Reply:
I will do that, or a version of it that clears when run through mypy.